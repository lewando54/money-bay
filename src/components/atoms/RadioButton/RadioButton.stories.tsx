/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { useState } from 'react'
import { StoryObj, Meta } from '@storybook/react-native'
import React from 'react'
import RadioButton from './RadioButton'
import { GOOD_GREEN } from '../../../styling/GlobalTheme.style'


interface IRadioButtonMockupProps {
    chosen: boolean
    color?: string
	onPress: () => void
}

function StatefulRadioButton({
    chosen=false,
    color=GOOD_GREEN,
    onPress
}: IRadioButtonMockupProps){
    const [isChosen, setIsChosen] = useState(chosen)

    return (
        <RadioButton chosen={isChosen} color={color} onPress={(id, group, chosenArg) => {setIsChosen(!chosenArg); onPress()}}/>
    )
}

const meta = {
    component: StatefulRadioButton,
    argTypes: {
        chosen: { control: 'boolean' },
        color: { control: 'color' },
        onPress: { action: 'Pressed' }
    }
} satisfies Meta<typeof StatefulRadioButton>

export default meta
type Story = StoryObj<typeof StatefulRadioButton>

export const Default: Story = {
    args: {
        chosen: false,
        color: GOOD_GREEN
    }
}

export const Chosen: Story = {
    args: {
        chosen: true,
        color: GOOD_GREEN
    }
}

export const ChosenCustomColor: Story = {
    args: {
        chosen: true,
        color: '#f00'
    }
}
