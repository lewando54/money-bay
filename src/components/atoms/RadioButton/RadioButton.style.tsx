import { StyleSheet } from 'react-native'
export default StyleSheet.create({
    container: {
        borderRadius: 100, 
        width: 18, 
        height: 18, 
        borderColor: '#CED6E1', 
        borderWidth: 2, 
        alignItems: 'center', 
        justifyContent: 'center'
    }
})
