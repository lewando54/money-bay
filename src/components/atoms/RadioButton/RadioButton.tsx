import React from 'react'
import { Pressable } from 'react-native'
import radioButtonStyle from './RadioButton.style'
import Animated, { useAnimatedStyle, useSharedValue, withTiming, Easing } from 'react-native-reanimated'
import { GOOD_GREEN } from '../../../styling/GlobalTheme.style'
import { useEffect } from 'react'

export interface IRadioButtonProps {
    id?: number
    group?: string
    chosen: boolean
    color?: string
    onPress: (id: number, group: string, chosen: boolean) => void
    testID?: string
}

export default function RadioButton({
    id=0,
    group='default',
    chosen=false,
    color=GOOD_GREEN,
    onPress,
    testID
}: IRadioButtonProps){
    const circleSize = useSharedValue(0)

    const circleAnimatedStyle = useAnimatedStyle(() => ({
        width: withTiming(circleSize.value, {duration: 100, easing: Easing.inOut(Easing.exp)}),
        height: withTiming(circleSize.value, {duration: 100, easing: Easing.inOut(Easing.exp)})
    }))

    useEffect(() => {
        if(chosen){
            circleSize.value += 10
        }
    }, [])

    const onPressHandle = () => {
        if(onPress){
            onPress(id, group, chosen)
        }

        if(chosen){
            circleSize.value -= 10
        }
        else {
            circleSize.value += 10
        }
    }

    return (
        <Pressable onPress={onPressHandle} hitSlop={{top: 10, bottom: 10, left: 10, right: 10}} style={radioButtonStyle.container} testID={testID}>
            <Animated.View style={[{backgroundColor: color, borderRadius: 100}, circleAnimatedStyle]}>
            </Animated.View>
        </Pressable>
    )
}
