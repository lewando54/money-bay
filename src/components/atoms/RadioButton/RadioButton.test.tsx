import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react-native'
import RadioButton from './RadioButton'
describe('RadioButton', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<RadioButton />)
    expect(tree).toMatchSnapshot()
  })
})
