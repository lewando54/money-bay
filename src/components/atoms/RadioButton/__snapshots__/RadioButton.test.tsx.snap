// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton should be chosen when passed true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={
    {
      "bottom": 10,
      "left": 10,
      "right": 10,
      "top": 10,
    }
  }
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderColor": "#CED6E1",
      "borderRadius": 100,
      "borderWidth": 2,
      "height": 18,
      "justifyContent": "center",
      "width": 18,
    }
  }
>
  <View
    animatedStyle={
      {
        "value": {
          "height": 0,
          "width": 0,
        },
      }
    }
    collapsable={false}
    style={
      {
        "backgroundColor": "#3EB290",
        "borderRadius": 100,
        "height": 0,
        "width": 0,
      }
    }
  />
</View>
`;

exports[`RadioButton should render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={
    {
      "bottom": 10,
      "left": 10,
      "right": 10,
      "top": 10,
    }
  }
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderColor": "#CED6E1",
      "borderRadius": 100,
      "borderWidth": 2,
      "height": 18,
      "justifyContent": "center",
      "width": 18,
    }
  }
>
  <View
    animatedStyle={
      {
        "value": {
          "height": 0,
          "width": 0,
        },
      }
    }
    collapsable={false}
    style={
      {
        "backgroundColor": "#3EB290",
        "borderRadius": 100,
        "height": 0,
        "width": 0,
      }
    }
  />
</View>
`;
